android/src/main/java/app/capgo/plugin/health/HealthPlugin.kt

```
@@ -429,10 +430,14 @@ class HealthPlugin : Plugin() {
             try {
                 val result = manager.queryAggregated(client, dataType, startInstant, endInstant, bucket, aggregation)
                 call.resolve(result)
+            } catch (e: SecurityException) {
+                Log.w("HealthPlugin", "Permission denied for aggregation: ${e.message}", e)
+                call.reject(e.message ?: "Permission denied for aggregated data.", "permission-denied", e)
             } catch (e: IllegalArgumentException) {
                 call.reject(e.message ?: "Unsupported aggregation.", null, e)
             } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)